### PR TITLE
Avoid including logger hrl

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: include/snabbkaffe.hrl [hunk 1]
```diff
@@ -212,7 +210,11 @@
 
 -else. %% SNK_COLLECTOR
 
--define(tp(LEVEL, KIND, EVT), ?LOG(LEVEL, EVT #{?snk_kind => KIND})).
+-define(tp(LEVEL, KIND, EVT),
+        logger:log(LEVEL, EVT#{ ?snk_kind => KIND
+                              , mfa => {?MODULE, ?FUNCTION_NAME, ?FUNCTION_ARITY}
+                              , line => ?LINE
```
Comment (k32) [inline include/snabbkaffe.hrl:216]: It would be nice to include process metadata as well.

File: Makefile [hunk 2]
```diff
@@ -23,3 +23,7 @@ $(CONCUERROR):
 	mkdir -p _build/
 	cd _build && git clone https://github.com/parapluu/Concuerror.git
 	$(MAKE) -C _build/Concuerror/
+
+.PHONY: xref
+xref:
```
Comment (k32) [inline Makefile:28]: This could be moved to the `compile` target, so xref runs all the time

**Review by k32 — COMMENTED** [review]
Thanks for the PR!

File: include/snabbkaffe.hrl (hunk 1 above)
Comment (zmstone) [reply include/snabbkaffe.hrl:216]: what's process metadata ?

Comment (k32) [reply include/snabbkaffe.hrl:216]: Logger stores some additional data in the process dictionary (see `http://erlang.org/doc/man/logger.html#set_process_metadata-1`). Logger's `?LOG` macro attaches the contents of the process metadata to all the logs emitted by the process, that could be used e.g. to filter log messages by domain and what not.

Comment (zmstone) [reply include/snabbkaffe.hrl:216]: I believe it's added for `logger:log` calls too.
https://github.com/erlang/otp/blob/OTP-23.2.6/lib/kernel/src/logger.erl#L1070-L1084
both `?LOG` macro and `logger:log` calls the same `log_allowed`

Comment (k32) [reply include/snabbkaffe.hrl:216]: You're right, I misremembered how the LOG macro works.

**Review by k32 — APPROVED** [review]

**Pull request merged**